module/users/helper.js

```diff
@@ -467,6 +467,63 @@ module.exports = class UsersHelper {
           });
         }
 
+        let totalCount = 0;
+        let mergedData = [];
+        //fetch all targeted programs to users
+        let targetedPrograms = await programsHelper.forUserRoleAndLocation(
+          requestedData,
+          "", // not passing page size
+          "", // not passing page number
+          "",
+          ["_id"],
+          programId
+        );
+        let targetedProgramIds = gen.utils.arrayOfObjectToArrayOfObjectIdInString(targetedPrograms.data);
+
+        // check current program is targeted or not
+        if(targetedProgramIds.length === 0){
+          let solutionIds = []
+          let importedSurveys = await surveyService.getStartedSurveys(token,programId)
+          importedSurveys = importedSurveys.result
+          let importedObservations = await surveyService.getStartedObservations(token,programId)
+          importedObservations = importedObservations.result
+          let importedProjects = await improvementProjectService.importedProjects(token,programId); 
```

Comment:
@ankitpws this will always return projects right? If so then why put type filter on Line 492?

Reply:
No it will check for each case it will check